### PR TITLE
feat(vue,nuxt): Add Vue and Nuxt SDK docs

### PR DESCRIPTION
**Pull request by wobsoriano**
> [!IMPORTANT]
> 🔎 Vue Previews:
>
> - https://clerk.com/docs/pr/1710/references/vue/overview (Overview)
> - https://clerk.com/docs/pr/1710/references/vue/migrating-from-vue-community-sdk (Migration guide)
>
> 🔎 Nuxt Previews:
>
> - https://clerk.com/docs/pr/1710/quickstarts/nuxt (Quickstart)
> - https://clerk.com/docs/pr/1710/quickstarts/overview#full-stack (Full stack entry)
> - https://clerk.com/docs/pr/1710/references/backend/overview
> - https://clerk.com/docs/pr/1710/references/nuxt/clerk-middleware (`clerkMiddleware` usage in Nuxt)
> - https://clerk.com/docs/pr/1710/references/nuxt/overview (Overview)
> - https://clerk.com/docs/pr/1710/references/nuxt/protecting-pages (Protecting pages)
> - https://clerk.com/docs/pr/1710/references/nuxt/read-session-data (Read session and user data)

### Explanation:

- We are launching an official Vue SDK for Clerk! 🎉
- We need to show how to migrate from existing [community SDK](https://vue-clerk.com) to the official one

### This PR:

- Is the base branch of other Vue SDK docs PRs
- Adds SDK References Overview page
- Adds community SDK to official SDK migration guide
- Sets the foundation for next PRs (client-side helpers, dead links for now)

### Checklist

- [x] I have clicked on "Files changed" and performed a thorough self-review
- [x] I have added the "deploy-preview" label and added the preview link(s) to this PR description
- [x] All existing checks pass

### CODE REVIEW

File: docs/references/vue/overview.mdx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+---
+title: Clerk Vue SDK
+description: Learn how to use Clerk to quickly and easily add secure authentication and user management to your Vue application.
+---
+
+Clerk makes it simple to add authentication to your Vue application. This documentation covers the capabilities and methods available from Clerk's Vue SDK.
+
+See the [quickstart](/docs/quickstarts/vue) to get started.
```
Comment (wobsoriano) [inline docs/references/vue/overview.mdx:8]: Quickstart guide PR [here](https://github.com/clerk/clerk-docs/pull/1709).

File: docs/manifest.json [hunk 2]
```diff
@@ -2427,6 +2427,74 @@
...
+                        {
+                          "title": "`useClerk()`",
+                          "wrap": false,
+                          "href": "/docs/references/vue/use-clerk"
+                        },
+                        {
+                          "title": "`useAuth()`",
+                          "wrap": false,
+                          "href": "/docs/references/vue/use-auth"
+                        },
```
Comment (wobsoriano) [inline docs/manifest.json:2466]: PR for the ff. pages [here](https://github.com/clerk/clerk-docs/pull/1711)

File: docs/manifest.json [hunk 3]
```diff
@@ -2427,6 +2427,74 @@
...
+                        {
+                          "title": "`useSignIn()`",
+                          "wrap": false,
+                          "href": "/docs/references/vue/use-sign-in"
+                        },
```
Comment (wobsoriano) [inline docs/manifest.json:2471]: PR for the ff. pages [here](https://github.com/clerk/clerk-docs/pull/1711)

File: docs/manifest.json [hunk 4]
```diff
@@ -2427,6 +2427,74 @@
...
+                        {
+                          "title": "`useSessionList()`",
+                          "wrap": false,
+                          "href": "/docs/references/vue/use-session-list"
+                        },
+                        {
+                          "title": "`useOrganization()`",
+                          "wrap": false,
+                          "href": "/docs/references/vue/use-organization"
+                        }
```
Comment (wobsoriano) [inline docs/manifest.json:2491]: PR for the rest of the pages [here](https://github.com/clerk/clerk-docs/pull/1712)

File: docs/references/vue/overview.mdx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+Clerk Vue provides a suite of [custom composables](/docs/references/vue/use-user). These composables give you access to the [`Clerk` object](/docs/references/javascript/clerk/clerk), and a set of useful helper methods for signing in and signing up.
+
+## Setting up Clerk Vue
+
+You need to create a Clerk application in your Clerk Dashboard before you can set up Clerk Vue. For more information, check out [the dedicated guide.](/docs/quickstarts/setup-clerk)
+
+Once a Clerk application has been created, you can install and start using Clerk Vue in your application. [The quickstart guide](/docs/quickstarts/vue) will have all the information you need to get started.
```
Comment (wobsoriano) [inline docs/references/vue/overview.mdx:19]: Quickstart guide PR [here](https://github.com/clerk/clerk-docs/pull/1709).

File: docs/references/vue/overview.mdx [hunk 6]
```diff
@@ -0,0 +1,19 @@
+---
+title: Clerk Vue SDK
+description: Learn how to use Clerk to quickly and easily add secure authentication and user management to your Vue application.
+---
+
+The Clerk Vue SDK is a wrapper around the Clerk JavaScript SDK. It is the recommended way to integrate Clerk into your Vue application.
+
+> [!IMPORTANT]
+> If you're using Nuxt, we recommend installing the `@clerk/nuxt` package since it's specifically built as a Nuxt module and includes backend integration. See the [Nuxt](/docs/references/nuxt/overview) documentation for more information.
```
Comment (wobsoriano) [inline docs/references/vue/overview.mdx:9]: This contains a dead link that will redirect to [this PR](https://github.com/clerk/clerk-docs/pull/1728)

**Comment by victoriaxyz** [comment]
Hey @wobsoriano! Here's the updates: https://github.com/clerk/clerk-docs/pull/1710/commits/fe61e86b176f3480b0c9f905ccbbf4dde73ac484 and https://github.com/clerk/clerk-docs/pull/1710/commits/1aec4f496c64f89c2a14337d7e2e8e419f6e335f

Update
- docs/references/vue/overview.mdx: reorganize and rewrite doc
- docs/references/vue/migrating-from-vue-community-sdk.mdx: refactor composable section into steps

Note
- docs/references/vue/migrating-from-vue-community-sdk.mdx: needs a link to clerk/vue doc: “official support for Vue”
- docs/references/vue/overview.mdx: 
- add link to “Nuxt docs” in important callout
- add custom composables methods (maybe)

File: docs/manifest.json (hunk 3 above)
Comment (victoriaxyz) [reply docs/manifest.json:2710]: I love how you broke these PRs out. Let's also add the relevant manifest links to each PR so we can preview while reviewing.

Comment (wobsoriano) [reply docs/manifest.json:2710]: Thanks! Are you talking about putting the client-side helper manifest links to this PR's description too?

Comment (victoriaxyz) [reply docs/manifest.json:2710]: To each PR that has new links in the manifest.

For example, I'm looking at `useAuth()` for this PR: https://github.com/clerk/clerk-docs/pull/1710. This way, I can view the helper functions locally. :)

Comment (wobsoriano) [reply docs/manifest.json:2710]: Ah gotcha!

**Comment by alexisintech** [comment]
I've removed the "prebuilt components" section from the overview - almost all of our SDKs have access to the prebuilt components, thats a given. I don't think we need to include that under "key features". key features is moreso for features are key to the vue sdk (and probably not found in other sdks)

**Review by victoriaxyz — APPROVED** [review]

**Comment by alexisintech** [comment]
needs nuxt docs in order to pass lint checks

**Comment by wobsoriano** [comment]
> needs nuxt docs in order to pass lint checks

Yes. Both Vue and Nuxt SDK will be [announced together](https://clerk-git-rob-eco-243-write-changelog.clerkstage.dev/changelog/2024-12-09-vue-and-nuxt-sdk) so I figured we merge them as one

File: docs/references/vue/use-auth.mdx [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+  - `Ref<string>`
+
+  The URL-friendly identifier of the user's active organization.
+
+  ---
+
+  - `signOut()`
+  - `Ref<(options?: SignOutOptions) => Promise<void>>`
+
+  A function that returns a promise that resolves when the current user is signed out. See the [reference doc](/docs/references/javascript/clerk/clerk#sign-out).
```
Comment (victoriaxyz) [inline docs/references/vue/use-auth.mdx:63]:
```suggestion
  A function that signs out the current user. Returns a promise that resolves when complete. See the [reference doc](/docs/references/javascript/clerk/clerk#sign-out).
```

File: docs/references/vue/use-session.mdx [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  - `Ref<boolean>`
+
+  A boolean that indicates whether Clerk has completed initialization. Initially `false`, becomes `true` once Clerk loads.
+
+  ---
+
+  - `isSignedIn`
+  - `Ref<boolean>`
+
+  A boolean that is set to `true` if the user is signed in.
```
Comment (victoriaxyz) [inline docs/references/vue/use-session.mdx:21]:
```suggestion
  A boolean that indicates whether a user is currently signed in.
```

File: docs/references/vue/use-sign-in.mdx [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+
+### `SignIn` status values
+
+The `status` property of the `SignIn` object can be one of the following values:
+
+<Properties>
+  - `complete`
+  - `string`
+
+  The user is signed in and the custom flow can proceed to `setActive()` to create session.
```
Comment (victoriaxyz) [inline docs/references/vue/use-sign-in.mdx:82]:
```suggestion
  The user is signed in and the custom flow can proceed to `setActive()` to create a session.
```

File: docs/references/vue/use-sign-up.mdx [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+<template>
+  <div v-if="!isLoaded">
+    <!-- Handle loading state -->
+  </div>
+
+  <div v-else>The current sign-up attempt status is {{ signUp?.status }}.</div>
+</template>
+```
+
+The `status` property of the `SignUp` object can be one of the following values:
```
Comment (victoriaxyz) [inline docs/references/vue/use-sign-up.mdx:76]:
```suggestion
### `SignUp` status values

The `status` property of the `SignUp` object can be one of the following values:
```

**Comment by victoriaxyz** [comment]
Commit to align Vue and React docs: https://github.com/clerk/clerk-docs/pull/1710/commits/7dd3ba615e8746bc95ebea37f6a4fb3c425c0fde

File: docs/quickstarts/vue.mdx [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+
+  You can control which content signed-in and signed-out users can see with Clerk's [prebuilt control components](/docs/components/overview#what-are-control-components). The following example creates a header using the following components:
+
+  - [`<SignedIn>`](/docs/components/control/signed-in): Children of this component can only be seen while **signed in**.
+  - [`<SignedOut>`](/docs/components/control/signed-out): Children of this component can only be seen while **signed out**.
+  - [`<UserButton />`](/docs/components/user/user-button): Shows the signed-in user's avatar. Selecting it opens a dropdown menu with account management options.
+  - [`<SignInButton />`](/docs/components/unstyled/sign-in-button): An unstyled component that links to the sign-in page or displays the sign-in modal.
+
+  ```vue {{ filename: 'src/App.vue', mark: [2, [6, 13]] }}
+  <script setup lang="ts">
```
Comment (victoriaxyz) [inline docs/quickstarts/vue.mdx:147]:
```suggestion
  <script>
```

**Review by alexisintech — APPROVED** [review]
thanks so much for all of the collaboration! 😸💖

**Review by mwickett — APPROVED** [review]

**Pull request merged**